src/Pelagos/Entity/DatasetSubmission.php

```diff
@@ -2373,6 +2373,9 @@ protected function updateDatasetStatus()
     {
         if ($this->getDataset() instanceof Dataset) {
             $this->getDataset()->setDatasetStatus($this->getDatasetStatus());
+            if ($this->getDatasetStatus() === Dataset::DATASET_STATUS_ACCEPTED) {
```

Comment:
This could be a re-submission, and an update being made via a new dataset review. This should only set the accepted date if the newest submission is being accepted, I think.

Reply:
Good catch!

Reply:
Fixed it 👍